ops/terraform/services/pipeline/modules/bfd_pipeline_manager/lambda_src/pipeline_manager.py

```diff
@@ -0,0 +1,374 @@
+import json
+import os
+import re
+import sys
+from dataclasses import asdict, dataclass
+from enum import Enum
+from typing import Any, Optional
+from urllib.parse import quote_plus, unquote
+
+import boto3
+from botocore.config import Config
+
+REGION = os.environ.get("AWS_CURRENT_REGION", "us-east-1")
```

Comment:
Probably need to support IDEMPOTENT_MODE here???

Reply:
This solution doesn't discern between running in non-`IDEMPOTENT` and `IDEMPOTENT` mode, so maybe this is something we should consider for the full implementation? Thanks for bringing this up.